contracts/evoting/mod_test.go

```diff
@@ -1204,7 +1206,306 @@ func TestCommand_AdminForm(t *testing.T) {
+	require.EqualError(t, err, getTransactionErr)
+
+	// Checking that providing a dummy data as argument, the form will not
+	// recognize it and won't be able to unmarshal it.
+	err = cmd.manageOwnersVotersForm(fake.NewSnapshot(), makeStep(t, FormArg, "dummy"))
+	require.EqualError(t, err, unmarshalTransactionErr)
+
+	// Checking that given a Blockchain that always returns an error with
+	// an Add cmd, it will not be able to retrieve the Form on the store.
+	err = cmd.manageOwnersVotersForm(fake.NewBadSnapshot(), makeStep(t, FormArg, string(dataAdd)))
+	require.ErrorContains(t, err, "failed to get key")
+
+	// Checking that given a Blockchain that always returns an error with
+	// a Remove cmd, it will not be able to retrieve the Form on the store.
+	err = cmd.manageOwnersVotersForm(fake.NewBadSnapshot(), makeStep(t, FormArg, string(dataRemove)))
+	require.ErrorContains(t, err, "failed to get key")
+
+	snap := fake.NewSnapshot()
+
+	// Checking that given the form set in the Snapshot which is invalid, then it
+	// will not be able to deserialize the Form to perform the command.
+	err = snap.Set(dummyFormIDBuff, invalidForm)
+	require.NoError(t, err)
+	err = cmd.manageOwnersVotersForm(snap, makeStep(t, FormArg, string(dataAdd)))
+	require.ErrorContains(t, err, deserializeErr)
+
+	// ====
+	// Now that we've performed check on the error, let's
+	// reset everything to perform the real test
+	// ====
+
+	err = snap.Set(dummyFormIDBuff, formBuf)
+	require.NoError(t, err)
+
+	// We retrieve the Admin Form from the ledger.
+	res, err := snap.Get(dummyFormIDBuff)
+	require.NoError(t, err)
+
+	message, err := formFac.Deserialize(ctx, res)
+	require.NoError(t, err)
+
+	form, ok := message.(types.Form)
+	require.True(t, ok)
+
+	// We check that now our dummy user is not a voter yet (return -1)
+	dummyUserVoterIndex, _ := form.GetVoterIndex(dummyUserID)
+	require.True(t, dummyUserVoterIndex == -1)
+
+	// We perform the Add command on the ledger
+	err = cmd.manageOwnersVotersForm(snap, makeStep(t, FormArg, string(dataAdd)))
+	require.NoError(t, err)
+
+	// We then check that the Add command was a success
+	res, err = snap.Get(dummyFormIDBuff)
+	require.NoError(t, err)
+
+	message, err = formFac.Deserialize(ctx, res)
+	require.NoError(t, err)
+
+	form, ok = message.(types.Form)
+	require.True(t, ok)
+
+	// We check that now our dummy user is an owner (return 0)
+	dummyUserVoterIndex, _ = form.GetVoterIndex(dummyUserID)
+	require.True(t, dummyUserVoterIndex == 0)
+
+	// Now let's remove it
+
+	// We perform below the command on the ledger
+	err = cmd.manageOwnersVotersForm(snap, makeStep(t, FormArg, string(dataRemove)))
+	require.NoError(t, err)
+
+	res, err = snap.Get(dummyFormIDBuff)
+	require.NoError(t, err)
+
+	message, err = formFac.Deserialize(ctx, res)
+	require.NoError(t, err)
+
+	form, ok = message.(types.Form)
+	require.True(t, ok)
+
+	// We check that now our dummy user is an owner (return 0)
```

Comment:
I think you forgot to change the comment here :)

Reply:
Good catch, updated 👍

Reply:
also, please change it to voter